### PR TITLE
Facility Booking Daily View

### PR DESCRIPTION
**Pull request by tinenhao**
https://trello.com/c/UQlAP3Ih
image

### CODE REVIEW

File: frontend/src/store/facilityBooking/action.ts [hunk 1]
```diff
@@ -13,6 +13,61 @@ export const SetCreateBookingError = (newError: string) => async (dispatch: Disp
...
+          }
+          updatedTB[hour] = t
+        }
+      }
+      dispatch(setSelectedDayBookings(updatedFB))
+      dispatch(setTimeBlocks(updatedTB))
+      dispatch(setIsLoading(false))
+    })
+
+  function resetTB() {
```
Comment (hui444) [inline frontend/src/store/facilityBooking/action.ts:60]: is there a need to make a funciton for this?

File: frontend/src/store/facilityBooking/action.ts [hunk 2]
```diff
@@ -13,6 +13,61 @@ export const SetCreateBookingError = (newError: string) => async (dispatch: Disp
...
+    '&endTime=' +
+    parseInt((adjustedEnd.getTime() / 1000).toFixed(0))
+
+  resetTB()
+  fetch(DOMAIN_URL.FACILITY + ENDPOINTS.FACILITY_BOOKING + '/' + querySubString, {
+    method: 'GET',
+    mode: 'cors',
+  })
+    .then((resp) => resp.json())
+    .then(async (res) => {
```
Comment (hui444) [inline frontend/src/store/facilityBooking/action.ts:36]: is there a need to use async for this?

File: frontend/src/store/facilityBooking/action.ts [hunk 3]
```diff
@@ -13,6 +13,61 @@ export const SetCreateBookingError = (newError: string) => async (dispatch: Disp
...
+            : new Date(updatedFB[i].endTime * 1000).getHours()
+        for (let hour = starthour; hour < endhour + 1; hour++) {
+          const t: TimeBlock = {
+            id: hour,
+            timestamp: updatedFB[i].startTime,
+            type: TimeBlockType.OCCUPIED,
+            ccaName: updatedFB[i].ccaName,
+            eventName: updatedFB[i].description,
+          }
+          updatedTB[hour] = t
```
Comment (hui444) [inline frontend/src/store/facilityBooking/action.ts:52]: could this and 45 be done in 1 line?

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (hui444) [inline yarn.lock:1]: please remove

**Review by hui444 — CHANGES_REQUESTED** [review]

**Review by hui444 — COMMENTED** [review]
do reference the design, the left side margin is different! 
image

File: frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+          />
+        </DatesContainer>
+      )
+    }
+  }
+
+  return (
+    <div>
+      <TopNavBarRevamp
+        onLeftClick={() => 'link to calendar page'}
```
[the remark this answers is not in the document]
Comment (tinenhao) [reply frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx:111]: i dont think it makes a difference, just need to wait for the calendar page to be up then can usehistory back to that page

File: frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
+import React, { useEffect } from 'react'
+import styled from 'styled-components'
+import { useHistory, useLocation } from 'react-router-dom'
+import 'antd-mobile/dist/antd-mobile.css'
+import 'antd/dist/antd.css'
```
[the remark this answers is not in the document]
Comment (tinenhao) [reply frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx]: fixed in [d6a507c](https://github.com/rhdevs/rhapp/pull/628/commits/d6a507cb58ee5539aaf8291b20d9d6a5e2113d24)

File: frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  }
+
+  return (
+    <div>
+      <TopNavBarRevamp
+        onLeftClick={() => 'link to calendar page'}
+        centerComponent={<TitleText>Calendar</TitleText>}
+        rightComponent={
+          <ButtonComponent state="primary" text="Book Facility" onClick={() => 'link to booking page'} size="small" />
```
[the remark this answers is not in the document]
Comment (tinenhao) [reply frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx:73]: i dont think it makes a difference, just need to wait for the calendar page to be up then can usehistory back to that page

File: frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  const location = useLocation<State>()
+  const { isLoading } = useSelector((state: RootState) => state.facilityBooking)
+  const selectedFacilityId = location.state.facilityId
+  const date = location.state.date
+
+  useEffect(() => {
+    dispatch(updateDailyView(date, selectedFacilityId))
+  }, [date])
+
+  function Dates(date: Date) {
```
[the remark this answers is not in the document]
Comment (tinenhao) [reply frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx]: fixed in [114bcce](https://github.com/rhdevs/rhapp/pull/628/commits/114bcce828dfb39e9af4768bf01bf609188bb866)

File: frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        </DatesContainer>
+      )
+    }
+  }
+
+  return (
+    <div>
+      <TopNavBarRevamp
+        onLeftClick={() => 'link to calendar page'}
+        centerComponent={<TitleText>Calendar</TitleText>}
```
[the remark this answers is not in the document]
Comment (tinenhao) [reply frontend/src/routes/FacilityBooking/ViewBookingDailyView/index.tsx]: fixed in [ca8c38d](https://github.com/rhdevs/rhapp/pull/628/commits/ca8c38db025eb18c7df784cab1efcd03f8eb4ad0)

File: frontend/src/store/facilityBooking/action.ts (hunk 1 above)
Comment (tinenhao) [reply frontend/src/store/facilityBooking/action.ts]: fixed in [7ccd566](https://github.com/rhdevs/rhapp/pull/628/commits/7ccd56628126b5a5e11de68a400e6587c66a2302)

File: frontend/src/store/facilityBooking/action.ts (hunk 2 above)
Comment (tinenhao) [reply frontend/src/store/facilityBooking/action.ts]: fixed in [dd8ee3e](https://github.com/rhdevs/rhapp/pull/628/commits/dd8ee3e66243b7621b53856d8bbc7980e506b869)

File: frontend/src/store/facilityBooking/action.ts (hunk 3 above)
Comment (tinenhao) [reply frontend/src/store/facilityBooking/action.ts]: fixed in [dd8ee3e](https://github.com/rhdevs/rhapp/pull/628/commits/dd8ee3e66243b7621b53856d8bbc7980e506b869)

File: yarn.lock (hunk 4 above)
Comment (tinenhao) [reply yarn.lock]: fixed in [6dc9ee2](https://github.com/rhdevs/rhapp/pull/628/commits/6dc9ee2c7232a5716e26229b75394d57a377edcc)

**Review by rldefa — APPROVED** [review]
lgtm

**Pull request merged**